### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by simon-p-r** [fishin/discuss#1]
Hello

If I can help with this project at all I will!  I like using hapi ecosystem and believe a CI would be great for node.

Thanks
Simon

**Comment by lloydbenson** [comment]
That'd be helpful!  The top level project is ficion.  Take a look at that README and let me know if it makes sense or needs more work.  Currently its somewhat hapi specific but it works in generic ways.  If its helpful I could start a gitter room or something if you want to collaborate / discuss ideas!

**Comment by simon-p-r** [comment]
Yes :+1:  I will do some reading and join a gitter room for discussions, I am relatively new to developing but eager to learn through experience.

**Comment by lloydbenson** [comment]
I look forward to collaborating!  I make a gitter room and added it to the README for both ficion repo and discuss.

**Comment by simon-p-r** [comment]
Ok I have got ficion working partially on my Windows box via git bash environment however I have made some comments based on instructions at ficion repo.
- Why use shell scripts and not node to launch ficion?  That way it will work cross platform.
- What permissions does the Github token require?
- Not clear what parameters needed changing in config file?
- Log folders missing or incorrect doesn't prevent process from launching, should it? 
- Set-up script could be run to configure this first type application is downloaded / run
- Tacklebox and gills plugins points to jobs, reel and users folders - what is this and do I need to create these?
- Basic user guide 

I am willing to help in whatever way as it is a cool project!

**Comment by lloydbenson** [comment]
Thanks for the feedback.  When doing this project I reduced scope a bit so I would actually accomplish something and I don't have a windows environment available.  So with that said:
1.  The shell scripts have some logic in them I didn't think was easy to do otherwise.  It generates an environment specific config, and backgrounds both the UI and the API processes.  Any suggestions on how to make this more generic to support windows?
2.  Permissions for github token.  Updated the README.  Is this sufficient now or do you think more is required?
3.  Interesting, I guess I wrongly assume you are familiar with how hapi plugins work (essentially all the options).  I think this is likely related to a setup script.  I had in the README to set something up.  Maybe a yeoman generator?  Thoughts on improvement?
4.  There is a bug against good-file for this with a help wanted tag:
   https://github.com/hapijs/good-file/issues/55
   It does complain but doesn't prevent app startup.
5.  See 3, but what do you think should be done, a yeoman generator?  I haven't used one, but if you have maybe you could offer some advice (or any other solution (cant use bash as you don't want it to not be supportable on windows)
6.  Those are just paths to store the data.  It's possible people want to separate out the directory structures in different ways.  It will auto create these but you still need to define these.  As part of the generator we could just make these relative?
7.  Yes, a user guide is a great idea, thoughts on some structure on an outline?

Followup, when you said you got it partially working, what are the specifics of what you didn't get going?

Thanks for all the great feedback and look forward to hearing about your ideas to make this more cross platform friendly.

**Comment by lloydbenson** [comment]
For 1 I changed npm start to start up lib/start.js.  It's just a start.  It won't background your process but it will start it up without hte bash script and run the confidence stuff it needs to.  To stop it you can just hit control c.  Let me know if thats sufficient.

For old behavior I setup npm run bash-start and run run bash-stop for myself ;)

Let me now if that is more windows friendly.

**Comment by lloydbenson** [comment]
For 6.  I added a couple of items in the README that should help.  Let me know if that clarifies it better until a more complete installer can be done.

**Comment by simon-p-r** [comment]
:tada: for new npm start, works a charm!

**Comment by lloydbenson** [comment]
Great.  Let me know once you get some directories for tacklebox created and your tacklebox_confidence.json updated to those directories if that works better.  I'll try to find out what folks are doing for config generation (I personally like to just edit config files but I realize some folks prefer more help there).

**Comment by simon-p-r** [comment]
I have used generators before which would be a quick way to generate files based on some user answered questions, it could create config files and create folders structures without much effort.

**Comment by lloydbenson** [comment]
Is "yo" a good use case here?  If so I can quickly research how to use it and do a npm run generate-config or something that calls that.

**Comment by simon-p-r** [comment]
Yes yo is easy to create a generator with

**Comment by lloydbenson** [comment]
create I'll read up on that and see how it goes.

**Comment by lloydbenson** [comment]
Cool, took a quick look, lots of libraries that are required for that.  I may just try to make one up using process first so I don't have to import so many lib dependencies.

**Comment by lloydbenson** [comment]
FYI i changed npm stop and start back to the original but you can use npm run start-simple for the non-bash way.  I have updated the README appropriately.
